src/NilPortugues/Laravel5/JsonApi/Controller/JsonApiController.php

```diff
@@ -45,10 +47,11 @@ public function index()
         $fields = $apiRequest->getFields();
         $sorting = $apiRequest->getSort();
         $included = $apiRequest->getIncludedRelationships();
-        $filters = $apiRequest->getFilters();
+        $filters = $apiRequest->getRawFilter();
```

Comment:
The getRawFilter has been implemented on a modified version of the php-json-api package that I've rolled which returns the entire and unmodified text of the filter parameter in the URL.   The method uses the $_SERVER['QUERY_STRING'] value to obtain the value as the ServerRequestInterface object's getQueryParams() method has run the values through urldecode which makes returning original filters provided by the user in URLs generated by the controller becomes near impossible.